exchanges/binance/binance.go

```diff
@@ -709,12 +709,13 @@ func (e *Exchange) SendHTTPRequest(ctx context.Context, ePath exchange.URL, path
 		return err
 	}
 	item := &request.Item{
-		Method:        http.MethodGet,
-		Path:          endpointPath + path,
-		Result:        result,
-		Verbose:       e.Verbose,
-		HTTPDebugging: e.HTTPDebugging,
-		HTTPRecording: e.HTTPRecording,
+		Method:             http.MethodGet,
+		Path:               endpointPath + path,
+		Result:             result,
+		Verbose:            e.Verbose,
+		HTTPDebugging:      e.HTTPDebugging,
+		HTTPRecording:      e.HTTPRecording,
+		MockDataSliceLimit: e.MockDataSliceLimit,
```

Comment:
Either this change needs to be replicated across the entire code for `MockDataSliceLimit`, or you drop these fields and use a context by value specifically for the mock system and then it will be constrained per specific request/nested requests for that call.

Reply:
I believe having this field is relevant wherever there is the HTTPRecording field in the codebase.

Reply:
Its still relevant in a context, so instead of having to worry about having those fields correctly set, we can just force a context check, then you can set up a global context with value with a struct with those fields with `func mock.Context(ctx, mock.Fields) ctx` for example or even just for one test function. What you have done is sufficient though. 🚀

Reply:
This is a very holy idea. Don't you think we may need to repeatedly call the **mock.WithMock()** in every unit test

Reply:
Yeah most likely need a global ctx, but defer this, can be done later when needed. especially for mocking websocket requests/responses.